nsxt/resource_nsxt_policy_ipsec_vpn_session.go

```diff
@@ -0,0 +1,494 @@
+				Optional:     true,
+			},
+			"compliance_suite": {
+				Type:         schema.TypeString,
+				Description:  "Connection initiation mode used by local endpoint to establish ike connection with peer site. INITIATOR - In this mode local endpoint initiates tunnel setup and will also respond to incoming tunnel setup requests from peer gateway. RESPOND_ONLY - In this mode, local endpoint shall only respond to incoming tunnel setup requests. It shall not initiate the tunnel setup. ON_DEMAND - In this mode local endpoint will initiate tunnel creation once first packet matching the policy rule is received and will also respond to incoming initiation request.",
+				ValidateFunc: validation.StringInSlice(IPSecVpnSessionComplianceSuite, false),
+				Optional:     true,
+			},
+			"connection_initiation_mode": {
+				Type:         schema.TypeString,
+				Description:  "Compliance suite.",
+				ValidateFunc: validation.StringInSlice(IPSecVpnSessionConnectionInitiationMode, false),
+				Optional:     true,
+				Default:      model.IPSecVpnSession_CONNECTION_INITIATION_MODE_INITIATOR,
+			},
+			"authentication_mode": {
+				Type:         schema.TypeString,
+				Description:  "Peer authentication mode. PSK - In this mode a secret key shared between local and peer sites is to be used for authentication. The secret key can be a string with a maximum length of 128 characters. CERTIFICATE - In this mode a certificate defined at the global level is to be used for authentication.",
+				ValidateFunc: validation.StringInSlice(IPSecVpnSessionAuthenticationMode, false),
+				Optional:     true,
+				Default:      model.IPSecVpnSession_AUTHENTICATION_MODE_PSK,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "Enable/Disable IPSec VPN session.",
+				Optional:    true,
+				Default:     true,
+			},
+			"psk": {
+				Type:        schema.TypeString,
+				Description: "IPSec Pre-shared key. Maximum length of this field is 128 characters.",
+				Optional:    true,
+			},
+			"peer_id": {
+				Type:         schema.TypeString,
+				Description:  "Peer ID to uniquely identify the peer site. The peer ID is the public IP address of the remote device terminating the VPN tunnel. When NAT is configured for the peer, enter the private IP address of the peer.",
+				Optional:     true,
+				ValidateFunc: validation.IsIPv4Address,
+			},
+			"peer_address": {
+				Type:         schema.TypeString,
+				Description:  "Public IPV4 address of the remote device terminating the VPN connection.",
+				Optional:     true,
+				ValidateFunc: validation.IsIPv4Address,
+			},
+			"tier0_id": {
+				Type:        schema.TypeString,
+				Description: "Unique identifier of the T0 resource.",
+				Optional:    true,
+			},
+			"locale_service": {
+				Type:        schema.TypeString,
+				Description: "Unique identifier of the Locale Service resource.",
+				Optional:    true,
+			},
+			"service_id": {
```

Comment:
To be consistent with provider convention, we should use `service_path` rather than `tier0_id`, `locale_service`, `service_id`. This way user can define service resource and reference it in subsequent VPN config, like here.